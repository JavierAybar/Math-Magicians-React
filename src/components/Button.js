import PropTypes from 'prop-types';

const Button = ({
  clickHandler,
  name,
  orange,
  grey,
  wide,
}) => {
  const handlerClick = () => clickHandler(name);

  const className = [
    'btn',
    orange ? 'orange' : '',
    grey ? 'grey' : '',
    wide ? 'wide' : '',
  ];

  return (
    <button className={className.join(' ').trim()} type="button" onClick={handlerClick}>{name}</button>
  );
};

Button.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  orange: PropTypes.bool,
  grey: PropTypes.bool,
  wide: PropTypes.bool,
};

Button.defaultProps = {
  orange: false,
  grey: false,
  wide: false,
};

export default Button;
