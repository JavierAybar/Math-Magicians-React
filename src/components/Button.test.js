import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button component', () => {
  test('renders correctly', () => {
    const clickHandlerMock = jest.fn();
    const { container } = render(<Button clickHandler={clickHandlerMock} name="Test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders with orange style', () => {
    const clickHandlerMock = jest.fn();
    const { container } = render(<Button clickHandler={clickHandlerMock} name="Test" orange />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders with grey style', () => {
    const clickHandlerMock = jest.fn();
    const { container } = render(<Button clickHandler={clickHandlerMock} name="Test" grey />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders as wide', () => {
    const clickHandlerMock = jest.fn();
    const { container } = render(<Button clickHandler={clickHandlerMock} name="Test" wide />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls click handler on button click', () => {
    const clickHandlerMock = jest.fn();
    const { getByText } = render(<Button clickHandler={clickHandlerMock} name="Test" />);

    fireEvent.click(getByText('Test'));
    expect(clickHandlerMock).toHaveBeenCalledWith('Test');
  });
});
