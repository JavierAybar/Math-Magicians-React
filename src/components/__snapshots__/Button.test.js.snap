// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders as wide 1`] = `
<button
  class="btn   wide"
  type="button"
>
  Test
</button>
`;

exports[`Button component renders correctly 1`] = `
<button
  class="btn"
  type="button"
>
  Test
</button>
`;

exports[`Button component renders with grey style 1`] = `
<button
  class="btn  grey"
  type="button"
>
  Test
</button>
`;

exports[`Button component renders with orange style 1`] = `
<button
  class="btn orange"
  type="button"
>
  Test
</button>
`;
